### DISCUSSION TITLE
Pico XMas Advent  Calendars Code

### DISCUSSION THREAD

**Discussion by gilesknap** [gilesknap/pico-xmas#4] [General]
This GitHub repo is all about writing code for the PiHut Pico Advent Calendars. 

If you have enjoyed the PiHut suggested code and want to try some more advanced Micro Python with your Advent Calendar Hardware then this is the right place. See the following links:

- [How to set up VSCode instead of Thonny](https://github.com/gilesknap/pico-xmas#pico-xmas)
- [Coding Ideas for the 2022 12 Days of Codemas Calendar](https://github.com/gilesknap/pico-xmas/blob/main/README2022.md#2022-the-12-projects-of-codemas)
- [Coding ideas for the 2023 Let It Glow Calendar](https://github.com/gilesknap/pico-xmas/blob/main/README2023.md#2023-let-it-glow)
- [My Final project with all the 2023 devices going at the same time](https://github.com/gilesknap/pico-xmas2)
If you have any questions, comments or ideas please post to this discussions page.

**Comment by gilesknap** [comment]
So, I have posted in the [Advent Calendar day #12](https://thepihut.com/blogs/raspberry-pi-tutorials/let-it-glow-maker-advent-calendar-day-12-dazzling-displays#article-561377214659-comments) comments that the code is finished. But that is no longer so! 

I've just received help from @Josverl @paulober who are the authors that make it possible to develop Python for the Pico inside of vscode.

I now have the means to add Python type hints to all of the code. This is something that all well polished Python should have because:
- it helps the code be self documenting
- it catches coding errors early

Here is a reasonable introduction to the concepts https://www.infoworld.com/article/3630372/get-started-with-python-type-hints.html.

Watch this repo for an update over the weekend to include full type hinting.

**Comment by ritchmct** [comment]
Thanks for this final project, Giles.

Great to have all the components working together. I just need to spend some time going through the code to see how you did it all.

The only small problem I encountered was the wiring on the SDA/SCL pins of the display. I think your wiring diagram may have them reversed.

Merry Christmas!
Ritchie

**Reply by gilesknap** [reply]
You are right I have fixed the issue in https://github.com/gilesknap/pico-xmas2/commit/d469fa0a3c95271fb16f6ba09e0e597fe8973d08.

Thanks for noticing that.
